modules/apps/commerce/commerce-pricing-web/src/main/resources/META-INF/resources/commerce_price_lists/cp_instance/js/AddPriceEntryForm.js

```diff
@@ -0,0 +1,347 @@
+	};
+
+	const onChangeUnitOfMeasureHandler = ({inputGroupId, unitOfMeasureKey}) => {
+		const inputGroup = getInputGroup(inputGroupId);
+		inputGroup.unitOfMeasureKey = unitOfMeasureKey;
+		setInputGroups([...inputGroups]);
+	};
+
+	const onRemoveInputGroupHandler = (id) => {
+		removeInputGroup(id);
+	};
+
+	const removeInputGroup = (id) => {
+		setInputGroups((inputGroups) =>
+			inputGroups.filter((inputGroup) => inputGroup.id !== id)
+		);
+	};
+
+	const submitForm = async (event) => {
+		event.preventDefault();
+
+		const form = event.currentTarget;
+
+		const error = form.querySelector('.has-error');
+
+		if (!error) {
+			const promises = inputGroups.map((inputGroup) => {
+				return AdminPricingResource.addPriceEntry(
+					inputGroup.priceListId,
+					{
+						price: inputGroup.price,
+						priceListId: inputGroup.priceListId,
+						skuId: cpInstanceId,
+						unitOfMeasureKey: inputGroup.unitOfMeasureKey,
+					}
+				)
+					.then((result) => {
+						result.inputGroupId = inputGroup.id;
+						result.success = true;
+
+						return result;
+					})
+					.catch((error) => {
+						error.success = false;
+
+						return error;
+					});
+			});
+
+			const results = await Promise.all(promises);
+
+			const validResults = results.filter((result) => result.success);
+
+			if (validResults.length === inputGroups.length) {
+				closeModal({
+					id: `${namespace}addPriceEntryDialog`,
+				});
+			}
+			else {
+				validResults.forEach((result) => {
+					removeInputGroup(result.inputGroupId);
+				});
+				Liferay.Util.openToast({
+					message: Liferay.Language.get(
+						'your-request-failed-to-complete-for-these-items'
+					),
+					title: Liferay.Language.get('error'),
+					type: 'danger',
+				});
+			}
+		}
+	};
+
+	return (
+		<ClayForm
+			className="add-price-entry-modal-form lfr-form-content m-2"
+			id={`${namespace}addPriceEntryForm`}
+			onSubmit={submitForm}
+		>
+			{inputGroups.map((inputGroup, index) => (
+				<ClayLayout.Sheet
+					className="add-price-entry-sheet"
+					key={`${namespace}inputGroup${index}`}
+					size="lg"
+				>
+					<ClayLayout.ContainerFluid>
+						<ClayLayout.Row justify="start">
+							<ClayLayout.Col
+								size={hasUnitOfMeasuresRef.current ? 4 : 5}
+							>
+								<label
+									htmlFor={`${namespace}commercePriceList${index}`}
+								>
+									{Liferay.Language.get('price-list')}
+
+									<span className="inline-item-after reference-mark text-warning">
+										<ClayIcon symbol="asterisk" />
+									</span>
+								</label>
+
+								<ClaySelectWithOption
+									id={`${namespace}commercePriceList${index}`}
+									name={`${namespace}commercePriceList${index}`}
+									onChange={({target}) => {
+										onChangePriceListHandler({
+											inputGroupId: inputGroup.id,
+											priceListId: target.value,
+										});
+									}}
+									options={priceLists}
+									required={true}
+								/>
+							</ClayLayout.Col>
+
```

Comment:
not sure this will be hidden if the feature flag of uom is disabled, can you check?

Reply:
The new modal is under UOM feature flag